joint_state_broadcaster:
  type: joint_state_broadcaster/JointStateBroadcaster
  ros__parameters:
    publish_rate: 500.0

eff_joint_trajectory_controller:
  type: joint_trajectory_controller/JointTrajectoryController
  ros__parameters:
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint
    interface_name: effort
    state_publish_rate: 500.0
    action_monitor_rate: 10.0
    stop_trajectory_duration: 0.5
    constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      shoulder_pan_joint:
        trajectory: 0.1
        goal: 0.1
      shoulder_lift_joint:
        trajectory: 0.1
        goal: 0.1
      elbow_joint:
        trajectory: 0.1
        goal: 0.1
      wrist_1_joint:
        trajectory: 0.1
        goal: 0.1
      wrist_2_joint:
        trajectory: 0.1
        goal: 0.1
      wrist_3_joint:
        trajectory: 0.1
        goal: 0.1
    gains:
      shoulder_pan_joint: {p: 20000,  d: 100, i: 10, i_clamp: 1}
      shoulder_lift_joint: {p: 20000,  d: 100, i: 20, i_clamp: 1}
      elbow_joint: {p: 10000,  d: 100, i: 10, i_clamp: 1}
      wrist_1_joint: {p: 500,  d: 1, i: 1, i_clamp: 1}
      wrist_2_joint: {p: 500,  d: 1, i: 1, i_clamp: 1}
      wrist_3_joint: {p: 10,  d: 0.1, i: 0, i_clamp: 1}

joint_group_effort_controller:
  type: effort_controllers/JointGroupEffortController
  ros__parameters:
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint
